### ISSUE TITLE
Clients cannot connect after upgrading server to 0.6.0

### ISSUE THREAD

**Issue by kevs-oc** [cloudradar-monitoring/rport#501]
Was on server version 0.5.0 on Ubuntu 18. Used the upgrade script to upgrade to 0.6.0. No errors were reported. Now clients cannot connect to the server, citing:

client: Connection error: websocket: bad handshake (Attempt: 10)

Tried rolling back the upgrade by copying the 0.5.0 version back to /usr/local/bin and restarting but the server stops logging after:

client-listener: Listening on 0.0.0.0:80...

**Comment by thorstenkramm** [comment]
@kevs-oc 
Just to check what has caused the trouble:
Download the 0.6.0 binary again to  a temp folder and just execute `./rportd --version` and `ldd ./rportd`.
Ubuntu 18.04 might not fulfill the glibc dependencies of rport 0.6.0 version.

**Comment by kevs-oc** [comment]
I rolled back to 0.5.0 and my old clients can connect however when I try to add a new client, it also gets the bad handshake error. Systemctl status shows rport is active/running and the rport.log shows just the handshake errors and ever increasing attempt intervals.

**Comment by thorstenkramm** [comment]
Are the new clients inside the same network as the old ones?
The bad handshake error is often caused by transparent proxies or firewalls doing virus scanning. Is there a transparent proxy on the network?
It must not be related to the server. Usually, you can connect 0.6.0 clients to older servers.

**Comment by kevs-oc** [comment]
And I take it back about the OS. I'm actually running on Ubuntu 20.04.3. Sorry about that. One more thing I noticed on the server side. When rportd starts, it must be calling guacd because I see it starting as well. I confirmed that rportd.conf has no reference to it and that the exe I'm running is 0.5.0.

**Comment by thorstenkramm** [comment]
The update to 0.6.0 has a dependency to guacd where 0.5.0 does not have it. You can remove guacd with `apt purge` or `dpkg`.
But if you are on Ubuntu 20.04 0.6.0 should run flawlessly. `ldd` confirmed all decencies are met.

**Comment by kevs-oc** [comment]
The new client is on the same network as an old client that is able to connect. This is lab network I am testing these new devices on and there is no transparent proxy running.

**Comment by kevs-oc** [comment]
After upgrading none of the existing clients could connect. The clients were a mix of 0.5.0 and 0.6.0.

**Comment by thorstenkramm** [comment]
After the upgrade, please check

- `systemctl status rportd`. Any errors?
- `netstat -tulpen`. Is rportd listening on the expected ports? 
- `/var/log/rport/rportd.log`. Any errors?
- If rportd does not come up, check for config syntax errors by executing in the foreground with `su - rport -s /bin/bash -c "rportd -c /etc/rport/rportd.conf"`
- Increase the log level in `rportd.conf` to debug, restart rportd and inspect the logs.

**Comment by kevs-oc** [comment]
I compared the rport.conf files from the working and non-working devices and found the server urls to be different. The newly installed clients are trying to reach my server using "https://servername.domain.com" while the previously installed clients that can connect are using "servername.domain.com:80". I changed a newly installed client to the same format and restarted the client. Now the client logs that there is a connection error: client id "blah" is already in use. The id referenced does not match the client name/password. Any thoughts?

**Comment by otto404-114** [comment]
I had a similar issue, changed the "https" to "http" in the client's rport.conf, with that workaround clients can connect. Seems to be a protocol mismatch.

**Comment by kevs-oc** [comment]
I did that and was able to connect but got the client id in use error. The new conf file that comes down has a new param called use_system_id which defaults to true. Since I imaged these appliances using the same OS image, they are getting the same machineid, which gets blocked by rportd. I set that param to false, and enabled the id value in the conf file as well as the new use_hostname param and name values. Now the new client can connect. 

I thought I had restored everything to 0.5.0 but did I miss the default conf file? Or does that come down from the provisioning server? If so, can I flag it to use the older version?

**Comment by thorstenkramm** [comment]
Yes. Clients connect over HTTP (without S). Encryption happens on application layer, not on the transport layer.

@kevs-oc 
The backup executed before the update process contains the old server config. The update will introduce new config options the old version doesn't understand. If you roll back to 0.5.0 you must restore the `rportd.conf`. 
There is no "default conf". It's all in the `rportd.conf`. Internal defaults are compiled into the binary.

**Comment by kevs-oc** [comment]
I double-checked my backup file and it does not contain a backup of the rportd.conf file. According to the upgrade script:
`# Create a backup
FOLDERS="/usr/local/bin/rportd /var/lib/rport /var/log/rport"
throw_info "Creating a backup of your RPort data. This can take a while."
throw_debug "${FOLDERS} will be backed up."
BACKUP_FILE=/var/backups/rportd-$(date +%Y%m%d-%H%M%S).tar.gz
require_pv
if is_available pv; then
  EST_SIZE=$(du -sb /var/lib/rport | awk '{print $1}')
  tar cf - $FOLDERS | pv -s $EST_SIZE | gzip > $BACKUP_FILE
else
  tar cvzf $BACKUP_FILE $FOLDERS
fi

throw_info "A backup has been created in $BACKUP_FILE"
`
that file is not backed up. 

I may be confused so please bear me out. Are you saying the conf file that the clients get is born from the rportd.conf file on the server? If so I did not see client section nor many of the params that the client has that the server does not.

**Comment by thorstenkramm** [comment]
@kevs-oc 
Damn, you are right. The upgrade does not include the rportd.conf in the backup. I'll update the script right now.

The client configuration is not related to the server configuration. The client configuration is "born" from the [sample configuration](https://github.com/cloudradar-monitoring/rport/blob/0.6.0/rport.example.conf) for a specific version. 

Client credentials are the only client-specific data. They are stored in a separated file, typically `/var/lib/rport/client-auth.json`
